### PR TITLE
 Set up Windows runner for unit tests CI job

### PR DESCRIPTION
**Pull request by garyttierney**
*NOTE*: still WIP.

## Check list

- [x] I have read through the [README](https://github.com/wfxr/minimap.vim/blob/master/README.md) (especially F.A.Q section)
- [x] I have searched through the existing issues or pull requests
- [ ] I have verified all existing unit tests pass (See the README on how to run)
- [ ] I have added new tests if appropriate
- [x] I have performed a self-review of my code and commented hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (when necessary)

## Description

This is an attempt to get the unit tests running under Windows in CI to hopefully help with testing issues like #109. There's a few kinks to iron out, so I've opened this as a draft until I get it passing. Ideally it'll remain mostly the same, with just the code-minimap stub and paths to neovim plugins changing.

## Type of change

- [ ] Bug fix
- [ ] New feature
- [ ] Improvement of existing features
- [ ] Refactor
- [ ] Breaking change
- [ ] Documentation change
- [x] CI / CD

## Test environment

- OS
    - [ ] Linux
    - [ ] Mac OS X
    - [ ] Windows
    - [ ] Others:
- Vim
    - [ ] Neovim: <Version>
    - [ ] Vim: <Version>

### CODE REVIEW

**Comment by garyttierney** [comment]
Ah, appears that CI won't run if it's a draft PR. I've marked it ready for review but left a note indicating it's still WIP.

**Comment by garyttierney** [comment]
This is running correctly now, but an issue in vim-testify prevents me overriding `g:testify#logger#type` to get good test result output. I'll try get that fix upstream.

**Comment by ZNielsen** [comment]
Just to make sure I understand the scope - this is intended to run the unit tests on Windows, but it does not fix the unit tests, correct?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -17,25 +17,49 @@ jobs:
...
         run: |
-          mkdir -p $HOME/.config/nvim/plugins
+          mkdir -p $NEOVIM_CONFIG_PATH/plugins
       - name: Create a fake code-minimap executable
         # Necessary for minimap.vim to start up
         run: |
           mkdir -p "$HOME/bin"
           ln -s /bin/echo "$HOME/bin/code-minimap"
           echo "$HOME/bin" >> $GITHUB_PATH
+        if: ${{ !contains(matrix.os, 'windows') }}
```
Comment (ZNielsen) [inline .github/workflows/ci.yml:46]: Do these `if` gates have to be at the bottom? It would be nice if it was the first part of the yaml section.

**Comment by garyttierney** [comment]
> Just to make sure I understand the scope - this is intended to run the unit tests on Windows, but it does not fix the unit tests, correct?

That's right. I'd hoped to look at them after finishing this but went down the rabbit hole of fixing vim-testify to get test results from it in CI. Unless you get to them before I wrap this branch up I'll take a look once I've got this producing good output.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (garyttierney) [reply .github/workflows/ci.yml]: I don't think so, I've moved them to the first key in the block. Let's see if GitHub accepts it.

**Review by ZNielsen — APPROVED** [review]

**Pull request merged**